src/test/ui/async-await/large_moves.rs

```diff
@@ -0,0 +1,24 @@
+#![deny(large_assignments)]
+#![feature(large_assignments)]
+#![move_size_limit = "1000"]
+// build-fail
+// only-x86_64
+
+// edition:2018
+
+fn main() {
+    let x = async { //~ ERROR large_assignments
```

Comment:
I think this is a particularly interesting instance of the problem (and perhaps a huge motivation for this lint).